### ISSUE TITLE
energy: add composition-only learned bandgap predictor for Matbench

### ISSUE THREAD

**Issue by Tristan-Stoltz-ERC** [Luminous-Dynamics/symthaea#2667]
## Problem

The current `symthaea-bandgap::BandgapPredictor` is not a valid drop-in learned predictor for `matbench_expt_gap`.

Its Random Forest feature vector includes:

```text
feature 16 = crystal system ordinal
```

and `BandgapPredictor::new()` trains on the curated Symthaea table using each material's actual crystal-system label.

Matbench `matbench_expt_gap` supplies composition only. Calling the existing predictor with `CrystalSystem::Unknown` only at benchmark inference would therefore create a train/inference distribution mismatch and would silently use a model whose training representation had structural information unavailable to benchmark candidates.

## Required V0 theorem

Add an explicitly composition-only learned predictor whose representation is identical in training and inference.

Acceptable V0 designs include:

1. remove the crystal-system feature from the learned feature vector; or
2. mask crystal system to one fixed value for **every training and inference sample**, proving the feature is constant/non-informative.

Do not infer crystal structure from benchmark truth, database lookups, candidate identity, or post-hoc hand labeling.

## Freeze the recipe

The learned-model identity should bind at least:

- exact exposed Symthaea training-snapshot digest;
- feature schema/version;
- explicit proof that crystal information is absent/masked symmetrically;
- RF tree count;
- max depth;
- minimum leaf/sample rule;
- deterministic RNG seed/algorithm;
- residual baseline identity/version;
- target-independent training procedure;
- exact implementation/environment subject through the qualification receipt.

Changing any recipe input should create a different model identity.

## Required prediction surface

Expose a composition-only API such as:

```text
predict(composition) -> prediction + uncertainty
```

with no crystal-system argument.

The Matbench screening adapter should consume #2653's retained truth-free composition rows directly and emit a `ScreeningRun` without opening benchmark truth.

## Evaluation discipline

The model may train on the frozen Symthaea curated training snapshot after #2653 excludes exact composition collisions from the Matbench screening universe.

However:

```text
exact-composition exclusion
!= globally clean holdout
```

Historical/manual knowledge, publication familiarity, hand-tuned baseline coefficients, external pretrained-model exposure, and near-neighbor leakage remain separate disclosures.

Do not tune RF hyperparameters, feature choices, target window, stopping rules, or candidate exclusions after reading Matbench truth outcomes.

## Cross-validation

The existing `BandgapPredictor::cross_validate()` returns only aggregate MAE/RMSE. V0 should preferably expose deterministic per-sample out-of-fold predictions/identities so internal qualification can establish that composition-only training behaves sensibly without using Benchmark Zero truth for tuning.

OOF evaluation of the Symthaea training table is model-development evidence; it is not a substitute for the external Matbench holdout.

## Acceptance regressions

At minimum:

1. public learned prediction API has no crystal-system input;
2. training path cannot access per-material crystal-system labels;
3. changing only crystal labels in a training fixture leaves model/predictions unchanged;
4. exact training-snapshot change changes model identity;
5. fixed recipe + fixed training snapshot is deterministic;
6. per-sample OOF prediction never trains on that exact sample;
7. screening candidate set equals #2653 retained universe exactly;
8. screening code has no `BandgapTruthSet` input;
9. benchmark truth cannot affect model fitting or ranking;
10. exact-composition overlap remains excluded before external screening.

## Relationship

- #1722: pinned official Matbench truth adapter
- #2653: truth-free retained composition universe
- #2666: composition-only legacy baseline + random-order null controls

This issue owns the first learned model that is actually representation-compatible with the composition-only benchmark.

**Comment by Tristan-Stoltz-ERC** [comment]
Implementation progress: the requested composition-only model and Matbench learned-screening adapter are now authored as stacked draft PRs #2674 and #2681.

#2674 adds symmetric composition canonicalization/crystal masking, equal-weight exact-composition group training, complete semantic RF recipe fields, and group-held-out OOF so same-composition polymorphs cannot cross folds.

#2681 binds that semantic recipe to #2653's exact exposed-training snapshot, carries the snapshot as a singleton `ScreeningMethodProvenance.training_slices` entry, consumes only the truth-free retained universe, and emits replayable learned `ScreeningRun` evidence without a `BandgapTruthSet` input.

Keep this issue OPEN. These are authored source/tests only. Closure requires exact-head pinned-toolchain format/check/test/strict-Clippy for the model + adapter and inherited Cargo.lock reconciliation. A queued workflow is not qualification.
